### PR TITLE
Allow users to select the bind address (ip) to use with `--bind`

### PR DESCRIPTION
**Pull request by emilk**
### What
Allow users to select the bind address (ip) to use with `--bind`

The default is still `0.0.0.0`, and still works (at least on my Mac).

I'm no network expert, so some of these names could probably be improved

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

PR Build Summary: https://build.rerun.io/pr/2159

### CODE REVIEW

File: crates/rerun/src/web_viewer.rs [hunk 1]
```diff
@@ -17,19 +17,21 @@ struct WebViewerSink {
...
 
-        let rerun_server = RerunServerHandle::new(rerun_rx, ws_port)?;
-        let webviewer_server = WebViewerServerHandle::new(web_port)?;
+        let rerun_server = RerunServerHandle::new(rerun_rx, bind_ip.to_owned(), ws_port)?;
+        let webviewer_server = WebViewerServerHandle::new(bind_ip, web_port)?;
 
         let web_port = webviewer_server.port();
         let server_url = rerun_server.server_url();
-        let viewer_url = format!("http://127.0.0.1:{web_port}?url={server_url}");
+        let viewer_url = format!("http://{bind_ip}:{web_port}?url={server_url}");
```
Comment (jleibs) [inline crates/rerun/src/web_viewer.rs:34]: Interesting. This looks problematic to me and I was surprised that it worked when using 0.0.0.0.

See: https://unix.stackexchange.com/questions/419880/connecting-to-ip-0-0-0-0-succeeds-how-why

Either way it means even if 0.0.0.0 works for localhost the url you generate will still be wrong: 0.0.0.0 most definitely won't work if you're trying to access from another computer.

I think the proper thing to do here, similar to how we handle the ephemeral port, is to expose, e.g.:
```
bound_ip = webviewer_server.ip()
```
and actually resolve that ip post bind.

**Review by jleibs — CHANGES_REQUESTED** [review]

File: crates/rerun/src/web_viewer.rs (hunk 1 above)
Comment (emilk) [reply crates/rerun/src/web_viewer.rs]: `127.0.0.1` doesn't work from another computer either.

I've made a commit where I ask the running server for its `local_addr` (its where we get the resolved port from). On my machine that still reports `0.0.0.0` which again, works locally but not from another computer (just like `127.0.0.1`).

Comment (jleibs) [reply crates/rerun/src/web_viewer.rs]: Yeah, I was just looking at your pushed change and noting the same thing.

File: crates/re_web_viewer_server/src/lib.rs [hunk 2]
```diff
@@ -247,22 +247,27 @@ impl WebViewerServer {
         Ok(())
     }
 
-    pub fn port(&self) -> WebViewerServerPort {
-        WebViewerServerPort(self.server.local_addr().port())
+    pub fn local_addr(&self) -> std::net::SocketAddr {
+        self.server.local_addr()
```
[the remark this answers is not in the document]
Comment (emilk) [reply crates/re_web_viewer_server/src/lib.rs:251]: Exactly - we could use something like https://crates.io/crates/local-ip-address for that, but that's not the point of this PR. The point of this PR is to let users select what bind addr to use. The default of `0.0.0.0` is reported as "You can connect to http://0.0.0.0` which is still _true_, at least on Mac. So as far as I can tell, this PR is still an improvement.

**Review by jleibs — APPROVED** [review]

**Pull request merged**